docs/troubleshooting-guide/sync-fingerprint-scan.md

```diff
@@ -1,85 +1,184 @@
 
+```sql
+SELECT TRUE
+FROM "your_database"
+WHERE 1 <> 1
+LIMIT 0
 ```
-SELECT *
-FROM table
-LIMIT 1
+
+If you’ve just set up a new database in Metabase, the initial sync query might still be running—--especially if you have a large database. To run the query, Metabase must:
+
+- successfully connect to your database, and
+- be granted permissions to query that database.
+
+If the connection is failing, or the database privileges are wrong, the sync query won't be able to run. This will block the _initial_ scan and fingerprinting queries as well.
+
+## Syncing new or updated tables and views
+
+1. Go to **Admin** > **Troubleshooting** > **Logs** to check the status of the sync.
+2. If the logs show you that the sync is failing on a specific table or view, make sure Metabase has the correct database privileges to query that table or view.
+3. Run a query against your database from the Metabase SQL editor to check for database connection or database privilege errors that aren't listed in the logs:
+
+    ```sql
+    SELECT *
+    FROM "your_database"."your_schema"."your_table_or_view"
+    LIMIT 1
+    ```
+4. [Manually re-sync](../databases/connecting.md#manually-syncing-tables-and-columns) the table or view.
+
+**Explanation**
+
+Metabase needs the correct database privileges to run the sync query against a given database, schema, and table (or view):
+
+```sql
+SELECT TRUE
+FROM "your_database"."your_schema"."your_table_or_view"
+WHERE 1 <> 1
+LIMIT 0
 ```
 
-Note that we only get the first 10,000 documents when scanning a MongoDB collection, so if you're not seeing some new fields, those fields might not exist in the documents we looked at. Please see [this discussion][metabase-mongo-missing] for more details.
+## Syncing tables with JSON records
+
+1. Go to **Admin** > **Databases** > **your database** > **Show advanced options**.
+2. Click **Disable "JSON unfolding"**
+3. Click **Save changes**.
+4. Click **Sync database schema**.
+
+**Explanation**
+
+Metabase will try to unfold JSON and JSONB records during the sync process, which can take up a decent chunk of query execution time. If you have a lot of JSON records, try disabling the automatic unfolding option to pull the sync out of slow-motion. Remember that you can follow the status of the sync from **Admin** > **Troubleshooting** > **Logs**.
+
+## Scanning
+
+1. Go to **Admin** > **Data Model**.
+2. Select the database and table.
+3. Go to the column you want to update, and click the **gear** icon.
+4. Click **Discard cached field values**.
+5. Click **Re-scan this field**.
+6. Go to **Admin** > **Troubleshooting** > **Logs** to follow the status of the scan.
+7. If you get an error during the scan process, try running the scan query against your database directly, and debug the query execution error from there. Check for:
+   - Schema and table privileges
+   - Recent schema or table updates
+   - Database-specific handling of nulls and numeric, timestamp, or boolean data types.
```

Comment:
None of these are relevant when dealing with scan. Privileges will fails for sync, schema changes will be picked up by sync, scanning only happens on string columns.

Reply:
Does it make sense to debug a scan query at all (should we remove step 7)?

Reply:
Remove it, or just refer to the logs to look for errors when scanning (not sync or fingerprinting, since it's different processes)